src/main/java/hillelee/doctor/DoctorRepository.java

```diff
@@ -0,0 +1,44 @@
+package hillelee.doctor;
+
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class DoctorRepository {
+    private List<Doctor> doctors = Collections.synchronizedList(new ArrayList<Doctor>());
+
+
+    public Collection<Doctor> findAll() {
+        return doctors;
+    }
+
+    public Optional<Doctor> findById(String id) {
+        return Optional.ofNullable(doctors.stream()
+                .filter(doctor -> doctor.getId().equals(id))
+                .findFirst()
+                .get());
+    }
+
+    public Doctor save(Doctor doctor) {
+        if (doctor.getId().isEmpty()) {
```

Comment:
if `id` 'is empty' it contains `null`, any method call on `null` (even `.isEmpty()`) will raise NullPointerException (NPE)